File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] sarr=new int[26];
        int[] carr=new int[26];
        for(int i=0;i<=25;i++){
            sarr[i]=0;
            carr[i]=0;
        }
        for(int i=0;i<s.length();i++){
            if(((int)s.charAt(i))>=97)
                sarr[(int)s.charAt(i)-97]++;
            else
                carr[(int)s.charAt(i)-65]++;
        }
        int hasextra=0;
        int sum=0;
        for(int i=0;i<=25;i++){
            sum+=sarr[i]/2+carr[i]/2;
            if(sarr[i]%2==1||carr[i]%2==1)
                hasextra=1;
        }
        return sum*2+hasextra;
    }
}

File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        hs = set()
        for i in s:
            if i in hs:
                hs.remove(i)
            else:
                hs.add(i)
        return len(s)-len(hs) + 1 if len(hs)>0 else len(s)